### ISSUE TITLE
Documentation error in the Metafuction tutorial

### ISSUE THREAD

**Issue by marehr** [seqan/seqan#1307]
In the Hint of [assignment 1](http://seqan.readthedocs.org/en/develop/Tutorial/Metafunctions.html#assignment-1):

> Use the Metafunctions Value to access the type of the elements in the container. Use the function value to assign the values.``

Firstly, [Value](http://docs.seqan.de/seqan/develop/?p=ContainerConcept#Value) is mentioned, but not used in the solution.

Secondly, [value](docs.seqan.de/seqan/develop/?p=RandomAccessContainerConcept#value) is marked as deprecated. Thus, the solution would be written as 

```c++
template<typename TCollection>
void swap(TCollection & items, int i, int j, int k) {
    TCollection tmps;
    resize(tmps, k);

    for(int x = 0; x < k; ++x) {
        tmps[x] = items[i + x];
    }

    for (int x = 0; x < k; ++x) {
        items[j + x] = items[i + x];
    }

    for (int x = 0; x < k; ++x) {
        items[i + x] = tmps[x];
    }
}
```

which uses no visible metafunctions and therefore the tutorial loses its purpose.

----------------

> If you want to search for metafunctions only you can do so by only selecting the metafunction category to the left of the search window at the online documentation.

This sentence makes no sense to me.

----------------

> The function is now quite generic allowing any container of type T. In addition we specify two positions that should be swapped (as integers which is not really generic, but it suffices for the demo) an the length of the swapped region. Now we can define a helper variable help, which can be of type T.

should be:
> demo) _and_ the length

**Comment by marehr** [comment]
She wanted it :smile:

**Comment by smehringer** [comment]
Fixed by #1520.

@marehr this issue can be closed.

**Comment by marehr** [comment]
Thanks!
